contracts/LotteryContractUpdated.sol

```diff
@@ -0,0 +1,135 @@
+            playerMap[msg.sender] = _player;
+            players.push(_player);
+            result = "Registered Successfully !!!";
+        }
+
+        return(result);
+        // new event to be send to owner
+    }
+    
+    function startLottery() public{
+        require (owner == msg.sender, "Only owner can open lottery process.");
+        state = LotteryState.Started;
+        
+    }
+
+    function stopLottery() public{
+        require (owner == msg.sender, "Only owner can open lottery process.");
+        state = LotteryState.Finished;
+        
+    }
+    
+    function buyLotteryTickets() public payable returns (string memory result) {
+        require(msg.value > .01 ether);
+        require(state == LotteryState.Started, "Lottery is not yet opened");
+        require(owner != msg.sender, "Lottery owner/organizer can't buy tickets.");
+        require(playerMap[msg.sender].isExist,"User not registered.");
+        
+        LotteryTicket memory ticket = LotteryTicket(tickets.length+1,1, msg.sender);
+        totalAmount = totalAmount + 1;
+        tickets.push(ticket);
+        result = "Ticket Bought Successfully !!!";
+        return(result);
+        //event to be sendout to owner
+    }
+    
+    function processLotteryWinners() public {
+        require(msg.sender == owner, "Only owner can execute ");
+        uint32 prizeMoney = totalAmount/2;
+        /*uint32 firstPrizeMoney = prizeMoney * 50/100;
+        uint32 secondPrizeMoney = prizeMoney * 30/100;
+        uint32 thirdPrizeMondy = prizeMoney * 20/100;
+        */
+        //Randomly select the numbers
+        
+        uint randomNumber = uint(keccak256(abi.encodePacked(now,  tickets.length))) % tickets.length;
+        LotteryTicket memory winnerTicket = tickets[randomNumber - 1];
+        winnerTicket.ticketOwner.transfer(address(this).balance);
+        winner.push(playerMap[winnerTicket.ticketOwner]);
+        
+        //Send notification
+        
+        state = LotteryState.NotStarted;
+    }
+    
+    function getLotteryState() view public returns (LotteryState _state ) {
+         //require(msg.sender == owner, "Only owner can execute ");
+         _state = state;
+    }
+
+    function isPlayerRegistered() view public returns (bool _status ) {
+        _status = playerMap[msg.sender].isExist;
+    }
+
+    function getLotteryTickets() view public returns (LotteryTicket[] memory _tickets ) {
+         require(msg.sender == owner, "Only owner can execute ");
+         return(tickets);
+    }
+    
+    function getPlayers() view public returns (Player[] memory _players ) {
+         require(msg.sender == owner, "Only owner can execute ");
+         return(players);
+    }
+    
+    function getWinner() view public returns (Player[] memory _players ) {
+        _players = winner;
+    }
+
+    //TODO - Not working 
+    function fetchMyTickets() view public returns (LotteryTicket[] memory _tickets ) {
```

Comment:
Would be good to have this in a working state too.

Reply:
I will do it today